src/System.IO.Compression/tests/ZipArchive/zip_ReadTests.cs

```diff
@@ -9,6 +9,22 @@ namespace System.IO.Compression.Tests
 {
     public class zip_ReadTests : ZipFileTestBase
     {
+        //[Fact]
+        //public static void ReadHugeFileWithoutEOCD()
+        //{
+        //    Diagnostics.Stopwatch sw = new Diagnostics.Stopwatch();
+
+        //    sw.Start();
+
+        //    Assert.Throws<InvalidDataException>(() =>
+        //    {
+        //        using FileStream fs = File.OpenRead(@"large.cab"); // 549 MB
```

Comment:
Do you really need actual files to test this, or can you just construct streams in memory?

Reply:
Let me see if I can create a huge file without an EOCD. If it's too much, I would prefer to save the file instead.

Reply:
Let me be more clear about this: please don't add a 500 MB file to our test data for a single test.  Cost/benefit is not worth it.  Test data is downloaded by all our contributors and not everyone has broadband.  Test data should be used sparingly.

Reply:
I don't know if it helps but you could make a fake Stream class that claims to be 1MB long, then unzip it, and make sure it doesn't call Read more than about 64K / n times, where n is the size of the reads.  My VSO bug has a stream implementation in the repro that just counts reads and seeks if you want a starting point.